web/src/containers/Withdraw/UserWithdrawalLimits.tsx

```diff
@@ -0,0 +1,122 @@
+                ctx.stroke();
+                ctx.beginPath();
+                ctx.strokeStyle = 'rgba(91, 100, 136, 0.24)';
+                ctx.lineWidth = 4;
+                ctx.arc(21, 24, 18, offset, 100);
+                ctx.stroke();
+            }
+        });
+    }, [usedWithdrawalLimitDay, currentUserWithdrawalLimitGroup, currencyId])
+
+    draw(usedWithdrawalLimitDay, currentUserWithdrawalLimitGroup?.limit_24_hour, 0);
+    draw(usedWithdrawalLimitMonth, currentUserWithdrawalLimitGroup?.limit_1_month, 1);
+
+    const renderTime = useCallback((period) => {
+        switch (period) {
+            case 'D':
+                return 'in 24 hours';
+            case 'M':
+                return 'in 1 month';
+            default:
+                return;
+        }
+    }, []);
+
+    const renderArcBlock = useCallback((period, estimatedValue, limit) => {
+        return (
+            <div className="cr-withdrawal-limits__group">
+                <canvas style={{ maxWidth: "46px" }} className="cr-withdrawal-limits__group-arc"></canvas>
+                <div className="cr-withdrawal-limits__group-info">
+                    <div className="cr-withdrawal-limits__group-info-usdt">
+                        <Decimal fixed={DEFAULT_FIAT_PRECISION} thousSep=",">{limit?.toString()}</Decimal>&nbsp;{GLOBAL_PLATFORM_CURRENCY}
+                        <span className="cr-withdrawal-limits__group-info-period">&nbsp;/{period}</span></div>
+                    <div className="cr-withdrawal-limits__group-info-currency">
+                        <Decimal fixed={fixed} thousSep=",">{estimatedValue?.toString()}</Decimal>&nbsp;{currencyId?.toUpperCase()}
+                        <span className="cr-withdrawal-limits__group-info-period">&nbsp;| {renderTime(period)}</span></div>
+                </div>
+            </div>
+        );
+    }, [price, fixed, usedWithdrawalLimitDay, usedWithdrawalLimitMonth, feeGroup, currentUserWithdrawalLimitGroup]);
+
+    return (
+        <React.Fragment>
+            <div className="cr-withdrawal-limits">
+                <h3>{translate('page.body.wallets.tabs.withdraw.content.withdrawal.limit')}</h3>
+                <span className="cr-withdrawal-limits__fee-group">{feeGroup.group}</span>
+                <OverlayTrigger
+                    placement="left"
+                    delay={{ show: 250, hide: 300 }}
+                    overlay={<Tooltip title="page.body.wallets.tabs.withdraw.content.withdrawal.limit.tip" />}>
+                    <div className="cr-withdrawal-limits__title-tip">
+                        <TipIcon />
+                    </div>
+                </OverlayTrigger>
+            </div>
+            {renderArcBlock('D', estimatedValueDay, +currentUserWithdrawalLimitGroup?.limit_24_hour - +usedWithdrawalLimitDay)}
+            {renderArcBlock('M', estimatedValueMonth, +currentUserWithdrawalLimitGroup?.limit_1_month - +usedWithdrawalLimitMonth)}

```

Comment:
it's not necessary to put all these props here as soon as these values are public in this scope

Reply:
you can use them directly in the function

Reply:
i just reuse the function, not need to make func to complicate with all conditions for handling Day and Month